include/zephyr/bluetooth/mesh/shell.h

```diff
@@ -25,15 +25,22 @@ extern "C" {
 		BT_MESH_HEALTH_PUB_DEFINE(_name,                               \
 					  BT_MESH_SHELL_CUR_FAULTS_MAX);
 
+struct bt_mesh_shell_target {
+	uint16_t local;
+	uint16_t dst;
+	uint16_t net_idx;
+	uint16_t app_idx;
+};
+
 /** @brief External reference to health server */
 extern struct bt_mesh_health_srv bt_mesh_shell_health_srv;
 
-/** @brief External reference to health client */
-extern struct bt_mesh_health_cli bt_mesh_shell_health_cli;
```

Comment:
I'm afraid we can't just remove this as `shell.h` is a public API (though not mentioned in Bluetooth Mesh Shell page). I think we should mark it as `__deprecated` and keep for 2 more releases (https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html?highlight=api+lifecycle#deprecated).

Reply:
Dose this meen that the bt_mesh_shell_health_cli instance in shell.c must be preserved as an empty placeholder?

Reply:
Actually, wait a minute, it seems like it can stay in shell.c (or shell_health_cli.c). The health cli shell commands will use the default instance or the one set by `instance set` command. `mesh_shell` can still use `bt_mesh_shell_health_cli` instance. But since `bt_mesh_shell_health_cli` was removed from the health cli shell commands then if `bt_mesh_shell_health_cli` is not used anywhere, it should be removed from the binary.

Reply:
Not 100% sure if i got your point. Should we preserve it in shell.c (without any CB handlers)?

Reply:
I think `bt_mesh_shell_health_cli` could stay in code without any changes. Unless we really want to get rid of it, but then it should be deprecated.